x/distribution/types/dec_coin.go

```diff
@@ -0,0 +1,102 @@
+package types
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Coins which can have additional decimal points
+type DecCoin struct {
+	Denom  string  `json:"denom"`
+	Amount sdk.Dec `json:"amount"`
+}
+
+func NewDecCoin(coin sdk.Coin) DecCoin {
+	return DecCoins{
+		Denom:  coin.Denom,
+		Amount: sdk.NewDec(coin.Amount),
+	}
+}
+
+// Adds amounts of two coins with same denom
+func (coin DecCoin) Plus(coinB DecCoin) DecCoin {
+	if !(coin.Denom == coinB.Denom) {
+		return coin
+	}
+	return Coin{coin.Denom, coin.Amount.Add(coinB.Amount)}
```

Comment:
This should be `DecCoin{}` not `Coin` right?

Reply:
oh yeah these types of error's are riddled throughout - I haven't gotten this to compile yet - still WIP on that front